pyaerocom/units/units.py

```diff
@@ -374,7 +381,7 @@ def convert(
             assert isinstance(other, Unit)
```

Comment:
Since you check already with if/else I would have 
```python

        if isinstance(other, Unit):
            to_unit = other
        elif isinstance(other, str):
            to_unit = Unit(str(other), **kwargs)
        else:
            throw AssertionError(f"other must be str|Unit, but is {type(other)}")